From 0627b75e315791f79e1e5aeaedb23fe92ac32e83 Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Fri, 18 Mar 2022 15:11:08 +0200
Subject: [PATCH 06/11] Documentation: Add ddr_gpr device-tree node bindings

Added the .yaml file which describes the bindings for the
ddr_gpr device-tree node.

Issue: ALB-8492
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../bindings/mfd/nxp,s32cc-ddr-gpr.yaml       | 37 +++++++++++++++++++
 1 file changed, 37 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/mfd/nxp,s32cc-ddr-gpr.yaml

diff --git a/Documentation/devicetree/bindings/mfd/nxp,s32cc-ddr-gpr.yaml b/Documentation/devicetree/bindings/mfd/nxp,s32cc-ddr-gpr.yaml
new file mode 100644
index 000000000000..18b668376954
--- /dev/null
+++ b/Documentation/devicetree/bindings/mfd/nxp,s32cc-ddr-gpr.yaml
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+# Copyright 2022 NXP
+%YAML 1.2
+---
+$id: "http://devicetree.org/schemas/mfd/nxp,s32cc-ddr-gpr.yaml#"
+$schema: "http://devicetree.org/meta-schemas/core.yaml#"
+
+title: NXP S32CC DDR_GPR regmap
+
+maintainers:
+  - Andrei Cherechesu <andrei.cherechesu@nxp.com>
+
+description: |
+  Device node that handles access by multiple devices to the
+  DDR_GPR registers.
+
+properties:
+  compatible:
+    items:
+      - const: nxp,s32cc-ddr-gpr
+      - const: syscon
+
+  reg:
+    maxItems: 1
+
+required:
+  - compatible
+  - reg
+
+additionalProperties: false
+
+examples:
+  - |
+    ddr_gpr: ddr_gpr@4007c600 {
+      compatible = "nxp,s32cc-ddr-gpr", "syscon";
+      reg = <0x4007c600 0x20>;
+    };
-- 
2.17.1

